raiden/transfer/mediated_transfer/initiator_manager.py

```diff
@@ -383,6 +390,100 @@ def handle_transferreroute(
     return TransitionResult(payment_state, events)
 
 
+def handle_transfer_reroute_light(
+    payment_state: InitiatorPaymentState,
+    state_change: ActionTransferRerouteLight,
+    channel_identifiers_to_channels: ChannelMap,
+    storage: SerializedSQLiteStorage
+) -> TransitionResult[InitiatorPaymentState]:
+
+    print("raiden/transfer/mediated_transfer/initiator_manager.py:396 running handle_transfer_reroute_light")
+
+    refund_transfer = state_change.refund_transfer
+
+    if not refund_transfer:
+        raise RaidenUnrecoverableError("Refund Transfer not found, state has changed with action "
+                                       "ActionTransferRerouteLight but refund_transfer param was not set")
+
+    store_refund_transfer = StoreMessageEvent(refund_transfer.message_identifier,
+                                              refund_transfer.payment_identifier,
+                                              1,
+                                              refund_transfer,
+                                              True,
+                                              LightClientProtocolMessageType.PaymentRefund,
+                                              refund_transfer.recipient)
+
+    try:
+        initiator_state = payment_state.initiator_transfers.get(state_change.transfer.lock.secrethash)
+        channel_identifier = initiator_state.channel_identifier
+        channel_state = channel_identifiers_to_channels[initiator_state.transfer.initiator].get(channel_identifier)
+    except KeyError:
+        return TransitionResult(payment_state, list())
```

Comment:
what is the logic behind this?

Reply:
I didn't made this logic, but what i could understand is that since this is a state transition we don't want to have any exceptions here, in case something is wrong we just return an empty transition list, this is replicated all over the place and i don't know why it was made like that.

Reply:
thanks for the explanation